src/main/java/net/snowflake/client/jdbc/SnowflakeUtil.java

```diff
@@ -493,6 +500,13 @@ public static String systemGetProperty(String property) {
     }
   }
 
+  /**
+   * System.getenv wrapper. If System.getenv raises an SecurityException, it is ignored and returns
+   * null.
+   *
+   * @param env the environment variable name.
+   * @return the environment variable value if set, otherwise null.
+   */
```

Comment:
Hey @sfc-gh-stakeda , could you please take a look at systemGetEnv function here to double check the functionality of it if fine, especially the re-throw of exception `throw ex`? I didn't fully follow the `systeGetProperty` as some of our code that used `System.getenv` rely on the catch of the exception if any. In order to make minimum change of our logic in stability period, I just re-throw the caught exception. (Btw, all existing places that used `System.getenv` are wrapped with `try-catch`)

Reply:
@sfc-gh-wshangguan , oh, I missed that the rethrow.... My intention was not to throw SecurityException. I'll update both of them.

Reply:
Just clarity. The primary goals of `systemGetProperty` and `systemGetEnv` is NOT to throw exception even if the environment doesn't allow to access the system property and environment variable, which are not requirements.

Reply:
Okay. Let me double that this won't break other code.

Reply:
@sfc-gh-stakeda , I feel like we need also change corresponding code in `FileCacheManager.java`. If not, in the case of can't retrieve environment variables & system properties, we will create cache dir with a wrong default path. E.g., on MacOS, we would get `current_path/Library/Caches/Snowflake/` instead of `~/Library/Caches/Snowflake/`

Reply:
you mean `homeDir`? Yes, let's check `homeDir` and return `this` if it is null.

Reply:
Yeah, sounds good! Just checked that other places should be fine.